src/hotspot/share/runtime/continuation.cpp

```diff
@@ -1130,106 +1134,96 @@ class Freeze : public FreezeBase {
     // assert(CodeCache::find_blob(*(address*)(top_sp - SENDER_SP_RET_ADDRESS_OFFSET)) == StubRoutines::cont_doYield_stub(), ""); -- fails on Windows
     assert(StubRoutines::cont_doYield_stub()->frame_size() == ContinuationHelper::frame_metadata, "");
     intptr_t* const stack_top     = top_sp + ContinuationHelper::frame_metadata;
-    const int       stack_argsize = _cont.argsize();
-    intptr_t* const stack_bottom  = _cont.entrySP() - ContinuationHelper::frame_align_words(stack_argsize);
+    intptr_t* const stack_bottom  = _cont.entrySP() - ContinuationHelper::frame_align_words(_cont.argsize());
 
     int size = stack_bottom - stack_top; // in words
 
     const int chunk_sp = chunk->sp();
     if (chunk_sp < chunk->stack_size()) {
-      size -= stack_argsize;
+      size -= _cont.argsize();
     }
     assert(size > 0, "");
 
     bool available = chunk_sp - ContinuationHelper::frame_metadata >= size;
     log_develop_trace(jvmcont)("is_chunk_available: %d size: %d argsize: %d top: " INTPTR_FORMAT " bottom: " INTPTR_FORMAT,
-      available, stack_argsize, size, p2i(stack_top), p2i(stack_bottom));
+      available, _cont.argsize(), size, p2i(stack_top), p2i(stack_bottom));
     DEBUG_ONLY(if (out_size != nullptr) *out_size = size;)
     return available;
   }
 
   template <bool chunk_available>
   bool freeze_fast(intptr_t* top_sp) {
-    assert(_thread != nullptr, "");
     assert(_cont.chunk_invariant(tty), "");
     assert(!Interpreter::contains(_cont.entryPC()), "");
     assert(StubRoutines::cont_doYield_stub()->frame_size() == ContinuationHelper::frame_metadata, "");
 
     // properties of the continuation on the stack; all sizes are in words
-    intptr_t* const stack_top     = top_sp + ContinuationHelper::frame_metadata;
-    const int       stack_argsize = _cont.argsize();
```

Comment:
I like this spelled out in the places it's used.  I wonder if _cont.argsize() is the stacksize of the stack chunk oops.  There are no comments really what it is.
cont_size seems to be the stack size of the entire continuation, ie the sum of all the stack chunk oops?  Dunno.  We're going to need some comments.

Reply:
_cont argsize is normally: inline int StackChunkFrameStream<frame_kind>::stack_argsize()
But it depends what we mean, a single intepreted frame must have the args from caller to be mounted.
But if caller is already mounted we already have those arguments.